cmd/mtnormalise.cpp

```diff
@@ -64,16 +66,25 @@ void usage ()
     + Option ("niter", "set the number of iterations. (default: " + str(DEFAULT_MAIN_ITER_VALUE) + ")")
     + Argument ("number").type_integer()
 
+    + Option ("value", "specify the (positive) reference value to which the summed tissue compartments will be normalised. "
+                       "(default: " + str(DEFAULT_NORM_VALUE, 6) + ", SH DC term for unit angular integral)")
+    + Argument ("number").type_float (std::numeric_limits<default_type>::min())
+
+    + Option ("balanced", "incorporate the per-tissue balancing factors into scaling of the output images "
```

Comment:
So, strongly against including this: I argued in the past why we *removed* it, and this was at length discussed among Dave and I, and later Rami, when we originally designed the original algorithm and later on redesigned it from the ground up.  I sadly don't have the time today to write up a very lengthy text on this (again), as I'm dealing with a few other urgent deadlines sadly, but wanted to mention this already now in an attempt to be open and clear about this.  The short, somewhat simplified version of this is essentially: this strategy (i.e. switching this option on) *breaks* the requirement of using a single set of (average, or other) response functions across a group / groups of subjects in quantitative (A)FD studies, such as an FBA and others.  If there is in fact a strong kind of (overall) "miscalibration" of the response functions, or more specifically their magnitude, relatively speaking for the different tissue types (e.g. the CSF response function is 10 times to large in signal magnitude or something, relative to the WM/GM ones), then the only way to deal with this is also "globally": I'm not sure if my words are entirely clear here, but what I consequentially mean by this, is that (in a certain way of looking at this) in principle the balance factors within this algorithm should be the same for all subjects that one wants to normalise with respect to each other.  But this is the annoying struggle that Dave and I originally discussed so long about: there is no simple or elegant way to have such a property in place in a context where this step would have to be able to run on individual subjects.  The latter desire originally mostly arose due to the Docker / BIDs pipelines kind of design, where Dave tried to avoid such points of "the data having to come together across subjects" as much as possible in the FBA pipeline.
So well, balance factors are still a necessary evil for the basic strategy of the `mtnormalise` algorithm, and they have to be used internally at some point to have that flat-looking (or flat-wanting-to-be) map that the bias field is fit to, but they are only and internal construct, and exporting it "into" the result breaks the very definition of the kind of operation `mtnormalise` aims to perform, which is "applying a single bias/"normalisation"-field globally to the dMRI data, represented as the set of tissue types being fed to the algorithm".  Changing internal contrasts to this input (e.g. among the tissue types relatively), and doing so differently for different subjects, defeats the normalisation purpose that `mtnormalise` specifically aims for.  We're already having more than trouble enough to get this normalisation thing (and "single unique set of responses" thing) across; so I'm strongly against introducing confusion in this.

As a side note, I suppose everyone here also understands that there *are* indeed limitations to `mtnormalise`, if you hadn't thought all of the above through yet.  We know / knew this right from the start though, all the way back to the original design, even before when we introduced the log domain into it; it's nothing new (just in case the above led to some surprises for anyone, while reading it...?).  If it wasn't clear yet in any way, as mentioned above, this is fundamentally unavoidable if all subjects are "independently" put through this process.  There are ways to mitigate this up to a certain extent, but they would typically be rather involved for your average user.  I most settings, these limitations also shouldn't have a biasing effect on the outcome of studies (there's a few rare exceptions; but they're rare, and there's ways to mitigate some limitations up to a certain specific extent there).

Well, looks like my short explanation got a bit lengthy after all...  Just to clarify: I know the above may come across as somewhat blunt here and there, but rest assured there's no bad or angry intentions here.  I don't have a lot of time to write this up now, and I'd rather be as clear as possible; so nuance may be lacking a bit for sure.  I will add a "request changes" kind of review to this pull request for now; **_again don't mean it in any offensive way_**; I fully appreciate the effort of trying to improve commands and options of course (and I'm happy about the debate on the re-categorisation that's going on in the other comments; this should lead to user interface improvements for sure). However, due to the story on quantifying 3-tissue CSD I'm working on with different groups in a few different studies now, that'll have text and theory submitted in the very short term, I feel quite strongly on not introducing confusion for users on this topic.  I may elaborate later, but this is all time I can spend on this today...

Reply:
OK, I get the rationale, and I _think_ I agree with it... @Lestropie, what specific use case did you have in mind? Was this for debugging? I note the balance factors were already written out as `INFO` level messages, so presumably even the `-check_factors` option is somewhat redundant - assuming it genuinely is only supposed to be inspected in the (hopefully) rare cases where we/users need to figure out whether something went funny? Likewise, if the `-balanced` option is only there to facilitate debugging, I think there's already enough info there to debug if need be - they're simple scale factors that can retrospectively be applied to each ODF, right? What I'm saying is that if this is something that you anticipate might be needed for other reasons, by all means let's consider including it, but otherwise I think I'd rather keep things lean (not to mention avoid users doing something they shouldn't, although that's less of a concern for me, we can always document clearly that this is _not_ to be used in practice).